### ISSUE TITLE
[Darwin] MTRDevice should support wildcard reads

### ISSUE THREAD

**Issue by jtung-apple** [project-chip/connectedhomeip#27025]
The current MTRDevice:
1. The read API requires nonnull endpointID, clusterID, and attributeID
2. The cache does not do wildcard search

This issue is to track support for wildcard reads.

**Comment by bzbarsky-apple** [comment]
@jtung-apple Is this a duplicate of https://github.com/project-chip/connectedhomeip/issues/23375?

**Comment by woody-apple** [comment]
Yup, but I closed that one, and this now has a use case. Adding to 1.3

**Comment by woody-apple** [comment]
This is complete.
